### DISCUSSION TITLE
Confusion about the example of the Azure OpenAI configuration

### DISCUSSION THREAD

**Discussion by goofansu** [karthink/gptel#156] [Q&A] [answered]
The Azure OpenAI example displays configuring multiple models for a single endpoint.

```elisp
(gptel-make-azure
 "Azure-1"
 :protocol "https"
 :host "RESOURCE_NAME.openai.azure.com"
 :endpoint
 "/openai/deployments/DEPLOYMENT_NAME/completions?api-version=2023-05-15"
 :stream t
 :models '("gpt-3.5-turbo" "gpt-4"))
```

As far as I know, Azure OpenAI requires to select a model when creating a deployment like the following:

![CleanShot 2023-12-26 at 16 14 30](https://github.com/karthink/gptel/assets/1143191/cb875a10-55d8-4b52-a3b1-00cc7513db16)

Is the example doable? Thanks

**Comment by karthink** [comment, marked as answer]
You're right, you'll need to use one gptel azure backend per deployment, and the `:models` field for the backend from your example above will be set to just `'("gpt-35-turbo")`.

**Reply by goofansu** [reply]
For anyone having the problem, here is an example for multiple models:

```elisp
(defvar azure-openai-api-host "YOUR_ENDPOINT") ; <deploy_name>.openai.azure.com
(defvar azure-openai-api-path "/openai/deployments/%s/chat/completions?api-version=2024-02-15-preview")
(defvar azure-openai-api-key (lambda () (auth-source-pass-get 'secret azure-openai-api-host))) ; customize to read your API key

(gptel-make-azure
 "Azure GPT-3.5"
 :host azure-openai-api-host
 :key azure-openai-api-key
 :endpoint (format azure-openai-api-path "gpt-35-turbo")
 :models '("gpt-3.5-turbo")
 :stream t)

(gptel-make-azure
 "Azure GPT-4"
 :host azure-openai-api-host
 :key azure-openai-api-key
 :endpoint (format azure-openai-api-path "gpt-4")
 :models '("gpt-4")
 :stream t)
```

**Reply by karthink** [reply]
Sorry, I lost track of this.  I'll make a single Azure backend work with multiple models, hang on.

**Reply by goofansu** [reply]
@karthink No worry, many thanks for the great package, it's embedded in my daily workflow. I added this comment just for other people's reference because I closed #158.
